### ISSUE TITLE
uif-contextual-menu - add support for advanced text option for menu item

### ISSUE THREAD

**Issue by s-KaiNet** [ngOfficeUIFabric/ng-officeuifabric#136]
As per #117 discussion, it would better if contextual menu will have an option to provide different options for item text.  
What I'm thinking about is to provide two options - one is leave it as it now - through the attribute, and the other option is an additional directive - `<uif-menu-item-text />`.  
`#1:`  

``` html
<uif-contextual-menu uif-is-open="isOpen">
  <uif-contextual-menu-item uif-text="'Delete'"></uif-contextual-menu-item>
</uif-contextual-menu>
```

`#2:`

``` html
<uif-contextual-menu uif-is-open="isOpen">
  <uif-contextual-menu-item>
    <uif-menu-item-text>
      <i class="ms-Icon ms-Icon--mail" aria-hidden="true"></i> <i>Icon</i>
    </uif-menu-item-text>
  </uif-contextual-menu-item>
</uif-contextual-menu>
```

Comments are very appreciated.  

I'm going to shift implementation on a later time and focus on other directive (`navbar` is waiting for me so long! :) ).

**Comment by andrewconnell** [comment]
I like the two options... no breaking change!

For option 2, do you need a nested container under the menu? If so, to stay in line with what we're talking about with `uif-command-bar` in #17 (see [this comment](https://github.com/ngOfficeUIFabric/ng-officeuifabric/issues/17#issuecomment-173787285))... specifically  instead of using `<uif-menu-item-text>` use `<uif-menu-item-content>`... so your option-2 above would be:

``` html
<uif-contextual-menu uif-is-open="isOpen">
  <uif-contextual-menu-item>
    <uif-menu-item-content>
      <uif-icon uif-type="mail" /> <span class="ms-fontColor-redDark">Hello World</span>
    </uif-menu-item-content>
  </uif-contextual-menu-item>
</uif-contextual-menu>
```

**Comment by s-KaiNet** [comment]
Good point, agreed.  
I need nested container, because menu may has sub-menu, in that case I need to distinguish sub menu from menu item text somehow. If text will have its own directive (`uif-menu-item-content`) that will be good.

**Comment by andrewconnell** [comment]
Gotcha... LGTM then

**Comment by s-KaiNet** [comment]
Can be closed because of #198
